Fallback service name (#2308)
open-telemetry/opentelemetry-dotnet-instrumentation: CHANGELOG.md

## [Unreleased](https://github.com/open-telemetry/opentelemetry-dotnet-instrumentation/compare/v0.6.0...HEAD)

> - Updated `ResourceConfigurator.cs` to set a fallback service name if one is not

Reviewer: The `CHANGELOG.md` is intended for users, the details of the implementation, e.g.: source file name, are, in general, not relevant for users. This should move to the `## Added` section, here is my wording suggestion (@theletterf is our expert here):

```suggestion
- Added a fallback service name if one is not
```

provided

> - `FallbackServiceName.cs` to retrieve a more helpful service name based on the

Reviewer: ```suggestion
- The fallback service name is based on the
```

Assembly name or ProcessName, if the application is .NET Framework and hosted

> in IIS this will be the SiteName/ApplicationVirtualPath

Reviewer: ```suggestion
in IIS this will be the SiteName/ApplicationVirtualPath.
```
